src/utils/ServiceUtil.java

```diff
@@ -0,0 +1,15 @@
+package utils;
+
+/**
+ * This class has utility methods for various services
+ **/
+public class ServiceUtil {
+    /**
+     Pick out the fragment or query part from a URL.
+     * @param url a URL possibly containing a query or a fragment part
+     * @return the query/fragment part
+     **/
+    public static String getUrlInfo(String url) {
+
```

Comment:
When you create a PR, do you make sure all the code is compiling? Looking at this class I don't think so.
For this first PR, I will review only if the code is compiling
From second PRRR onwards, I will review only if code is compiling and there are unit tests.
@lccodes FYI

Reply:
add return null to all